### PR TITLE
Add support to pickup prometheus-plugin-prometheus-sd config context

### LINKED ISSUE #120: Pass down prometheus scraping information in the config context
Hello,

We would like to use this plugin in some kind of "self service manner" netbox wise. Meaning that we would essentially use one http_sd_config (technically two for vms and server in our case) with something like: `/api/plugins/prometheus-sd/devices/?tags=prom-autoconfig` and then use config context to configure metrics_path, ports etc.

The idea is to have something similar as what we used to configure Kubernetes (essentially before or if prometheus-operator is not used) with annotation on pods like: `prometheus.io/scheme`, `prometheus.io/path`, `prometheus.io/port`.

I propose to put something like that in the config context:
```yaml
prometheus-plugin-prometheus-sd:
  - metrics_path: /not/metrics
    port: 4242
    scheme: https
  - port: 4243
```

And then the plugin would discover this and use it for the http sd json essentially. Of course I am happy to work on the implementation, in case that's something that could be included!

### PR DESCRIPTION
**Pull request by MrFreezeex**
## Describe your changes
Add the ability to automatically pickup config context with the following scheme:
```
prometheus-plugin-prometheus-sd:
  - metrics_path: /not/metrics
    port: 4242
    scheme: https
  - port: 4243
```

To do that I had add a custom list serializer to "duplicate" one entry to the number of entries of the array `prometheus-plugin-prometheus-sd` in the config context. The new serializer also inject a new variable that is used later on to the target and the labels.

Also add the ability in netbox >= 3.5 to restrict the API of this plugin to only list operations

## Issue ticket number and link
Fixes #120 
## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.

### CODE REVIEW

**Comment by FlxPeters** [comment]
I just started the review. One thing is validation. Given i add a invalid config context like:

```json
{
    "prometheus-plugin-prometheus-sd": "foo"
}
```
This breaks the whole API. Please add a schema validation to make sure the broken config is ignored in this case. 
Best would be a test case for that.

**Comment by FlxPeters** [comment]
As soon as we have proper error handling in place this is ready for merge. Thanks for your contribution.

I will add some more complex examples soon so people can better understand how to use this plugin.
This will also include a example for this case.

File: netbox_prometheus_sd/api/utils.py [hunk 1]
```diff
@@ -103,3 +103,14 @@ def extract_custom_fields(obj, labels: LabelDict):
             # Complex types are rendered as json
             else:
                 labels["custom_field_" + key.lower()] = json.dumps(value)
+
+def extract_prometheus_sd_config(obj, labels):
```
Comment (FlxPeters) [inline netbox_prometheus_sd/api/utils.py:107]: In the example you have a ` port: 4242` defined. This seams to be ignored here. Is this intended?

**Comment by MrFreezeex** [comment]
Hi @FlxPeters, thanks for the review, indeed I forgot about malformed types :(. The new versions should handle this type of things and I also added a few tests for those.

**Comment by FlxPeters** [comment]
Looks good to me. I will merge it and push a pre-release so you and others can test it.
Give me a hit if you are happy with the outcome and i will release it to a major version.

**Pull request merged**